.travis.yml

```diff
@@ -1,122 +1,159 @@
-       export PATH=$PATH:/Users/travis/Library/Python/2.7/bin
+     # Force OS X to use the correct python version. This is only because
+     # Travis' support for `language: python` on OS X is too buggy.
+     if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+       PYTHON_CONFIGURE_OPTS="--enable-framework" pyenv install $PYTHON
+       # This part is (apparently) quite important! As per the docs it makes
+       # pyenv aware that some executables have been installed/removed. Without it
+       # `pyenv global` won't be able to find the version of Python we're asking for.
+       pyenv rehash
+       # Printing the info about pyenv's python version before and after the change.
+       # We should see something like `system` before and `3.6.0` after.
+       pyenv global
+       pyenv global $PYTHON
+       pyenv global
+       # Manually update the PATH as advised in the pyenv docs.
+       PATH="$(pyenv root)/shims:$PATH"
+       # OS X build have `language: generic` so `TRAVIS_PYTHON_VERSION` is undefined
+       # at first.
+       TRAVIS_PYTHON_VERSION=$PYTHON
+     fi
+  - |
+     if ! which pip; then
+       python -m easy_install pip
+     fi
+  - |
+     # Check that `python` is actually the version we want
+     # Parses `TRAVIS_PYTHON_VERSION` and throws away the patch level as Travis only
+     # saves the major and minor versions.
+     _WE_WANTED=$(echo "$TRAVIS_PYTHON_VERSION" | sed -E 's/([0-9]+\.[0-9]+)($|\.[0-9]+)/\1/')
+     # Ask python itself what its major and minor versions are
+     _WE_GOT=$(python -c 'import sys; print(".".join(map(str, sys.version_info[:2])))')
+     if [[ $_WE_WANTED != $_WE_GOT ]]; then
+       # The `&& false` here is needed to produce the error code indicating failure, because
+       # we don't want to proceed if Python is not installed properly.
+       echo "Something went wrong during python installation: $_WE_WANTED != $_WE_GOT" && false
      fi
-  # make sure CXX is correctly set
-  - if [[ "${COMPILER}" != "" ]]; then export CXX=${COMPILER}; fi
 
-  # Get all required Python packages
-  - pip install numpy scipy matplotlib networkx pytest mpi4py numdifftools --user
+  # Check that 'Python.h' can be found, i.e. that we didn't forget to install development files.
+  # Otherwise we'll run into trouble when trying to use pybind11.
+  - find $(python -c 'from distutils import sysconfig; print(sysconfig.get_python_inc())') -name 'Python.h'
+
+install:
+  - cmake --version
+  - python --version
+  - python -m pip install -U pip
+  - python -m pip --version
+  - python -m easy_install -U setuptools
+  - python -m pip install numpy networkx pytest numdifftools
+  # TODO(twesterhout): Remove the following line when #109 is merged.
+  - python -m pip --no-cache-dir install mpi4py
+  - python -m pytest --version
+  - NETKET_CMAKE_FLAGS="$CMAKE_FLAGS" python -m pip install -v .
 
-  # show OS/compiler version
-  - uname -a
-  - $CXX --version
+script:
+  - python -c 'from mpi4py import MPI'
```

Comment:
Is this a leftover?

Reply:
No, this line is meant to remain until we remove the dependency on `mpi4py`. It helps with narrowing down MPI-related problems. If you insist, I can remove it of course, but would prefer not to.

Reply:
I agree that it's good to have this. If MPI fails independently of netket it is good to see that early on in the tests.